src/main/java/functions/VisitorFunction.kt

```diff
@@ -19,6 +23,21 @@ fun createVisitorFunction(node: ASTNode): VisitorFunction {
         is Sibling -> siblingFunction(node.siblings.map { createVisitorFunction(it) })
         is BreadthClosure -> breadthClosureFunction(createVisitorFunction(node.expression))
         is Union -> unionFunction(node.expressions.map(::createVisitorFunction))
+        is Check -> checkFunction(createVisitorFunction(node.expression))
         else -> throw IllegalArgumentException("This ASTNode subtype is not supported.")
     }
 }
+
+fun mergeList(list: List<List<List<TargetTreeNode>>>): List<List<TargetTreeNode>> {
+    if (list.size <= 1) {
+        return list.first()
+    }
+    val mergeList = mergeList(list.drop(1))
```

Comment:
Maybe we shouldn't use `.drop()` as it creates a copy of the sub list and as I understand `.subList()` does something similar creating a view of the original list.

This might create some trouble if we try to change the sub list obtained by `.subList()` but we are not doing it here and doing some tests I found out it ends up giving us the same result.

```suggestion
    val mergeList = mergeList(list.subList(1, list.size))
```

Reply:
Okay I'm going to change that, Thanks